src/lib/list/list.ts

```diff
@@ -50,6 +63,347 @@ export class MdList {
+  get disableRipple() { return this._disableRipple; }
+  set disableRipple(value: boolean) { this._disableRipple = coerceBooleanProperty(value); }
+
+  @ContentChildren(MdLine) _lines: QueryList<MdLine>;
+
+  /** Whether the label should appear after or before the checkbox. Defaults to 'after' */
+
+  @Input() checkboxPosition: 'before' | 'after' = 'after';
+
+  /** Whether the checkbox is disabled. */
+  get disabled() {
+    return this._disabled;
+  }
+  @Input('disabled')
+  set disabled(value: any) {
+    this._disabled = coerceBooleanProperty(value);
+  }
+
+  @Input('value')
+  get value() { return this._value; }
+  set value( val: any) { this._value = coerceBooleanProperty(val); }
+
+  @Input('selected')
+  get selected() { return this._selected; }
+  set selected( val: boolean) { this._selected = coerceBooleanProperty(val); }
+
+  /** Emitted when the option is focused. */
+  onFocus = new EventEmitter<MdSelectionListOptionEvent>();
+
+  /** Emitted when the option is selected. */
+  @Output() select = new EventEmitter<MdSelectionListOptionEvent>();
+
+  /** Emitted when the option is deselected. */
+  @Output() deselect = new EventEmitter<MdSelectionListOptionEvent>();
+
+  /** Emitted when the option is destroyed. */
+  @Output() destroy = new EventEmitter<MdSelectionListOptionEvent>();
+
+  constructor(private _renderer: Renderer2,
+              private _element: ElementRef,
+              @Optional() public selectionList: MdSelectionList,) {
+    this._isSelectionList = !!selectionList;
+  }
+
+
+  ngAfterContentInit() {
+    this._lineSetter = new MdLineSetter(this._lines, this._renderer, this._element);
+  }
+
+  ngOnDestroy(): void {
+    this.destroy.emit({option: this});
+  }
+
+  toggle(): void {
+    if(this._disabled == false) {
+      this._selected = !this._selected;
+      if(this._selected == true) {
+        this.selectionList.selectedOptions.select(this);
+      }else {
+        this.selectionList.selectedOptions.deselect(this);
+      }
+    }
+  }
+
+  /** Allows for programmatic focusing of the option. */
+  focus(): void {
+    this._element.nativeElement.focus();
+    this.onFocus.emit({option: this});
+  }
+
+  /** Whether this list item should show a ripple effect when clicked.  */
+  isRippleEnabled() {
+    return !this.disableRipple && this._isSelectionList
+      && !this.selectionList.disableRipple;
+  }
+
+  _handleFocus() {
+    this._hasFocus = true;
+    this._renderer.addClass(this._element.nativeElement, 'mat-list-item-focus');
+  }
+
+  _handleBlur() {
+    this._renderer.removeClass(this._element.nativeElement, 'mat-list-item-focus');
+  }
+
+  /** Retrieves the DOM element of the component host. */
+  _getHostElement(): HTMLElement {
+    return this._element.nativeElement;
+  }
+}
+
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-selection-list, mat-selection-list',
+  host: {'role': 'listbox',
+    '[attr.tabindex]': '_tabIndex',
+    'class': 'mat-selection-list',
+    // Events
```

Comment:
No need for category comments inside `host`; the different syntax makes it clear enough